engine/proofview.ml

```diff
@@ -264,7 +264,7 @@ module Monad = Proof
 
 (** [tclZERO e] fails with exception [e]. It has no success. *)
 let tclZERO ?(info=Exninfo.null) e =
-  if not (CErrors.noncritical e) then
+  if not (CErrors.noncritical e || e = CErrors.Timeout) then
```

Comment:
This is very dangerous style, you must use a pattern-matching instead.

Reply:
I've changed this to pattern-matching style, but the result is a bit awkward. Or did you have something else in mind.